File: Prob3.js

//determine the largest prime factor under a number

var test_number = 600851475143; //target number
var largest = 0;


function isPrime(num) {
 
    
//initialize isPrime value as true
    
    var isPrime = true;

// determine if isPrime is false by checking whether it is divisible
    //by each value under it, up to its square root
    
   for (var i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) {isPrime = false}
   }

   return isPrime;

}

// loop through numbers up to sqrt of target to see if they 
// divide into target. Then check to see if it's a prime number

for (var j = 1; j < Math.sqrt(test_number); j++) {
    if (test_number % j === 0) {
        if (isPrime(j)) {
            
            largest = j;
        }
    }
    
}
  
console.log(largest);

File: Prob3.rb
=begin
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
=end

def is_prime(n)
    test_result = true
    @t = 2
    upper = Math.sqrt(n).floor
   
        while @t <= upper do
           
            if (n % @t) == 0
                test_result = false
               
            end

            @t = @t+1
        end
    return test_result
end

 
outer_limit = 600851475143
i = 1
largest = 0

while i < Math.sqrt(outer_limit) do
    if outer_limit % i == 0  #is it a factor
        if is_prime(i) 
            #puts i #is it prime
            largest = i
        end
    end
    i = i + 1
end
puts largest


    
        